src/Paket.LoadingScripts/LoadingScriptsGenerator.fs

```diff
@@ -225,11 +110,69 @@ module ScriptGeneratingModule =
 
       dependencies
       |> Map.map (fun groupName packages ->
-        let packagesOrGroupFolder =
+        let packagesOrGroupFolder, groupName =
           match groupName.GetCompareString () with
-          | "main"    -> packagesFolder 
-          | groupName -> Path.Combine(packagesFolder.FullName, groupName) |> DirectoryInfo
-        generateFSharpScripts dependenciesFile lockFile packagesOrGroupFolder packages
+          | "main"    -> packagesFolder, None
+          | groupName -> Path.Combine(packagesFolder.FullName, groupName) |> DirectoryInfo, Some groupName
+        generateScripts getScriptName scriptGenerator framework dependenciesFile packagesOrGroupFolder groupName packages
         )
       |> ignore
-      
\ No newline at end of file
+
+  let generateFSharpScript (input: ScriptGenInput) =
+    let packageName = input.PackageInstallModel.PackageName
+    let packageFolder =
+      Path.Combine (input.PackagesOrGroupFolder.FullName, packageName.GetCompareString())
+      |> DirectoryInfo
+
+    let depLines =
+      input.DependentScripts
+      |> Seq.map (fun s -> sprintf "#load \"../%s\"" (s.Replace("\\", "/")))
+
+    let toRelative = (Path.GetFullPath >> (fun f -> f.Substring(packageFolder.FullName.Length + 1)))
+
+    let dllFiles =
+      if packageName.GetCompareString().ToLowerInvariant() = "fsharp.core" then
+        Seq.empty
+      else
+        getDllFilesWithinPackage input.Framework input.PackageInstallModel
+        |> Seq.map toRelative
+
+    let dllLines =
+      dllFiles
+      |> Seq.map (fun dll -> sprintf "#r \"%s\"" (dll.Replace("\\", "/")))
+
+    depLines
+    |> fun lines -> Seq.append lines dllLines
+    |> fun lines -> Seq.append lines [ sprintf "printfn \"%%s\" \"Loaded %s\"" (packageName.GetCompareString()) ]
+
+  let generateFSharpScriptsForRootFolder =
+    let getScriptName (framework: FrameworkIdentifier) (package: PackageName) = sprintf "Include_%s_%s.fsx" (package.GetCompareString()) (framework.ToString())
```

Comment:
most of paket related files are all lowercase (significant for linux I guess)

I really think we should have something like this:

packages/include-scripts/{platform}/include.PackageName.fsx

package name identical to the actual nuget package identifier so might contains upper case characters, but maybe lower case that one too?